src/algebra/group/type_tags.lean

```diff
@@ -13,14 +13,54 @@ variables {α : Type u} {β : Type v}
 def additive (α : Type*) := α
 def multiplicative (α : Type*) := α
 
-instance [inhabited α] : inhabited (additive α) := ⟨(default _ : α)⟩
-instance [inhabited α] : inhabited (multiplicative α) := ⟨(default _ : α)⟩
+/-- Reinterpret `x : α` as an element of `additive α`. -/
+def to_add (x : α) : additive α := x
+
+/-- Reinterpret `x : additive α` as an element of `α`. -/
+def of_add (x : additive α) : α := x
```

Comment:
This should probably be in the additive namespace so that field notation works.

Reply:
What rename do you propose?

Reply:
`additive.to_mul` is the best that I can think of now.

Reply:
Just to be sure that I understand correctly: do you propose `__root__.to_mul`, `additive.to_mul`, `__root__.to_add` and `multiplicative.to_add`? If we want to avoid polluting the `__root__` namespace, we can use `additive.of_mul`, `additive.to_mul`, `multiplicative.of_add` and `multiplicative.to_add`.

Reply:
I don't think @gebner is worried about pollution, but rather wants to enable writing `x.to_mul` for `x : additive foo`.

Reply:
@jcommelin That's right.  This is the definition I'm proposing:
```lean
def additive.to_mul (x : additive α) : α := x
```

Reply:
Then we should move `to_mul` to `multiplicative.of_add` because otherwise we'll have `__root__.to_mul` and `additive.to_mul`, and it will be hard to tell whether we go `α → multiplicative α` or `additive α → α`.

I think that I'll make a version of this PR that doesn't add `irreducible` but adds `to_mul` and `of_mul` + `simp` lemmas.

Reply:
The naming of this function was just a suggestion.  I still think that this PR makes the right choice in making `multiplicative` and `additive` irreducible.

Reply:
It leads to quite a few build failures. I want to merge `to_mul`/`of_mul`/`to_add`/`of_add` before fixing these failures.